app/src/main/java/com/hedvig/app/feature/swedishbankid/sign/SwedishBankIdSignDialog.kt

```diff
@@ -0,0 +1,177 @@
+
+import android.content.Intent
+import android.graphics.Color
+import android.graphics.drawable.ColorDrawable
+import android.net.Uri
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.compose.animation.animateContentSize
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Surface
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.ComposeView
+import androidx.compose.ui.platform.ViewCompositionStrategy
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.core.os.bundleOf
+import androidx.fragment.app.DialogFragment
+import androidx.lifecycle.flowWithLifecycle
+import com.hedvig.app.R
+import com.hedvig.app.feature.settings.MarketManager
+import com.hedvig.app.ui.compose.HedvigTheme
+import com.hedvig.app.util.extensions.canOpenUri
+import com.hedvig.app.util.extensions.toArrayList
+import com.hedvig.app.util.extensions.viewLifecycle
+import com.hedvig.app.util.extensions.viewLifecycleScope
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+import org.koin.android.ext.android.inject
+import org.koin.androidx.viewmodel.ext.android.viewModel
+import org.koin.core.parameter.parametersOf
+
+class SwedishBankIdSignDialog : DialogFragment() {
+    private val model: SwedishBankIdSignViewModel by viewModel {
+        parametersOf(
+            requireArguments().getString(AUTO_START_TOKEN)
+                ?: throw IllegalArgumentException(
+                    "Programmer error: Missing AUTO_START_TOKEN in ${this.javaClass.name}"
+                ),
+            requireArguments().getStringArrayList(QUOTE_IDS)
+                ?: throw IllegalArgumentException(
+                    "Programmer error: Missing QUOTE_IDS in ${this.javaClass.name}"
+                )
+        )
+    }
+    private val marketManager: MarketManager by inject()
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?,
+    ) = ComposeView(requireContext()).apply {
+        setViewCompositionStrategy(ViewCompositionStrategy.DisposeOnLifecycleDestroyed(viewLifecycleOwner))
+        setContent {
+            val viewState by model.viewState.collectAsState()
+            HedvigTheme {
+                SwedishBankIdSignDialog(
+                    text = stringResource(
```

Comment:
Think it makes sense to extract this to a separate function with viewState as param

Reply:
Something like `getTextByViewState(viewState: ViewState)`?

Reply:
Something like that yeah!